instrumentation/kafka/kafka-clients/kafka-clients-common/library/README.md

```diff
@@ -0,0 +1,233 @@
+# Kafka Metrics
+
+The Kafka client exposes metrics via `org.apache.kafka.common.metrics.MetricsReporter` interface.
+OpenTelemetry provides an implementation that bridges the metrics into OpenTelemetry.
+
+To use, configure `OpenTelemetryKafkaMetrics` with an OpenTelemetry instance
+via `OpenTelemetryKafkaMetrics#setOpenTelemetry(OpenTelemetry)`, and include a reference to this
```

Comment:
should we make this instrumentation rely on GlobalOpenTelemetry instead?

IIRC we've discussed in the past that SPI-based instrumentations (which also don't support injecting OpenTelemetry instance) will probably leverage GlobalOpenTelemetry (cc @anuraaga)

Reply:
Hmm, the JDBC driver instrumentation (which is kinda similar in its usage) does use `GlobalOpenTelemetry`.

Reply:
This isn't quite an SPI since you need to explicitly include a reference to the class in the producer or consumer client config. 

If possible, I think it's better to provide options that don't require global initialization, considering we explicitly advise against using the global unless required. Maybe we can provide a fallback where if `setOpenTelemetry` isn't called before the first usage, we initialize with `GlobalOpenTelemetry.get()`?

Reply:
this is just an instrumentation-specific "global OpenTelemetry"?

I guess the question is, should each instrumentation that relies on a global (statically accessible) OpenTelemetry instance have its own "global OpenTelemetry", or should they use `GlobalOpenTelemetry`?

there's definitely pros and cons for each approach, I just want to have a consistent answer to this question

Reply:
Discussed and determined we should take advantage of `MetricsReporter#configure(Map<String, ?>)` to inject the instance of OpenTelemetry in as a config property. The latest commit reflects this change.